### ISSUE TITLE
Make log entropy function out of CountVectorizer

### ISSUE THREAD

**Issue by titipata** [titipata/science_concierge#12]
- see https://github.com/scikit-learn/scikit-learn/blob/51a765a/sklearn/feature_extraction/text.py#L973

**Comment by titipata** [comment]
Here is the preliminary code implemented https://en.wikipedia.org/wiki/Latent_semantic_indexing, can you check @daniel-acuna?

``` python
import scipy.sparse as sp

def log_entropy_vectorizer(docs):
    """
    perform log entropy vectorizer as stated in 
    ref: https://en.wikipedia.org/wiki/Latent_semantic_indexing
        g_i = 1 + sum (p_ij * log(p_ij))/log(n)
    """
    X = CountVectorizer().fit_transform(docs)
    n_samples, n_features = X.shape
    gf = np.array(X.sum(axis=0)).ravel() # count total number of words
    P = (X * sp.spdiags(1./gf, diags=0, m=n_features, n=n_features)) # probability matrix
    p = P.data
    P.data = 1 + (p * np.log(p) / np.log(n_samples))
    g = np.array(P.sum(axis=0)).ravel()
    X.data = np.log(1 + X.data)
    G = sp.spdiags(g, diags=0, m=n_features, n=n_features)
    return X * G
```

**Comment by titipata** [comment]
@daniel-acuna, I tested the log-entropy function and it gives worse performance compared to normal LSA. I tried both log-entropy with l2 normalization and without normalization. This [paper](http://webpages.ursinus.edu/akontostathis/KontostathisHICSSFinal.pdf) also reported worse performance of LSA with log-entropy on Medical abstract. I don't have intuition yet why entropy can be worse in our case.

**Comment by daniel-acuna** [comment]
lets see those plots!

**Comment by titipata** [comment]
**note** we fix the sum of log-entropy now at 103e7880607f1b6f7b02697ca9b5d3c290dd49f8, thanks @daniel-acuna
